src/pagination/pagination.ts

```diff
@@ -0,0 +1,93 @@
+import {Component, EventEmitter, Input, Output, NgFor} from 'angular2/angular2';
+
+@Component({
+  selector: 'ngb-pagination',
+  directives: [NgFor],
+  template: `
+    <nav>
+      <ul class="pagination">
+        <li [class.disabled]="!hasPrevious()">
+            <a aria-label="Previous"  (click)="selectPage(page-1)">
+              <span aria-hidden="true">&laquo;</span>
+              <span class="sr-only">Previous</span>
+            </a>
+        </li>
+
+        <li *ng-for="#pageNumber of pages" [class.active]="pageNumber === page">
+          <a (click)="selectPage(pageNumber)">{{pageNumber}}</a>
+        </li>
+
+        <li [class.disabled]="!hasNext()">
+          <a aria-label="Next" (click)="selectPage(page+1)">
+            <span aria-hidden="true">&raquo;</span>
+            <span class="sr-only">Next</span>
+          </a>
+        </li>
+      </ul>
+    </nav>
+  `
+})
+export class NgbPagination {
+  private _collectionSize = 0;
+  private _page = 0;
+  private _pageSize = 10;
+  pages: number[] = [];
+
+  @Input()
+  set page(newPage: number | string) {
+    this._selectPage(parseInt(`${newPage}`, 10));
+  }
+
+  get page() : number | string {
+    return this._page;
+  }
+
+  @Input()
+  set collectionSize(newSize: number | string) {
+    this._collectionSize = parseInt(`${newSize}`, 10);
+    this._updatePages();
+  }
+
+  get collectionSize() : number | string {
+    return this._collectionSize;
+  }
+
+  @Input()
+  set pageSize(newSize: number | string) {
+    this._pageSize = parseInt(`${newSize}`, 10);
+    this._updatePages();
+  }
+
+  get pageSize() : number | string {
+    return this._pageSize;
+  }
+
+  @Output() pageChange = new EventEmitter();
+
+  hasPrevious(): boolean { return this._page > 1; }
```

Comment:
Ah no no, I mean `_page`. Change it to:

`hasPrevious(): boolean { return this.page > 1; }`

If an internal variable like `_page`, `_pageSize` has a set/get, it is better to always use them, because they could have extra logic that needs to be applied.

So if the `get page() { }` does extra logic in the future our code will be prepared to use that new logic. Apart, it is better to not having to code the _ all the time.

That is a classic convention on C# so I guess that Typescript will follow it too.